### PR TITLE
Compile rust bindings to Android

### PR DESCRIPTION
**Pull request by laurenspriem**
This PR is a first attempt at fixing [compilation of the rust bindings to Android](https://github.com/unum-cloud/usearch/issues/443).

Unfortunately simply adding an if statement for Android did not solve it for me, so more commits will be necessary.

### CODE REVIEW

**Comment by ashvardanian** [comment]
Hi @laurenspriem! Thank you for your contribution!
I'm having hard time getting that error, but face another one - `cargo` not being able to find the `algorithm` header.
How did you configure your environment?

**Comment by laurenspriem** [comment]
Hey @ashvardanian Appreciate the quick responses!
At this point I'm pretty convinced that all of these errors, including the first one that started this PR, are just due to incorrectly configured environments. Nothing is probably wrong with the actual code, so I'll close this PR soon I think. I'm sorry for dragging you into this!

As for configuration, you first need to install Android NDK as [per description](https://developer.android.com/studio/projects/install-ndk) or let [this project](https://github.com/bbqsrc/cargo-ndk) do it for you. But even using that project I was still getting compilation errors, which were only fixed for me when I used [this other package](https://github.com/cross-rs/cross) specifically for configuring for cross compilation. At which point I realized there's nothing wrong with the usearch code.

tldr; configuration is tricky for compiling Rust to Android it seems.

**Comment by ashvardanian** [comment]
If you have a minimal example of how it works with Android, I'd love to include a link into the README, similar to how we mention  [SwiftSemanticSearch](https://github.com/ashvardanian/SwiftSemanticSearch) on the main page. I'm sure many people would find your snippets useful 🤗

**Comment by laurenspriem** [comment]
I'll first focus on porting usearch to flutter using the rust bindings. Once that's working well, I'll make it into a flutter package and see if I can add a short description and link to the README :+1:
